src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -358,7 +358,15 @@ static void GenerateXamlCodeBehind(XamlProjectItem? xamlItem, Compilation compil
 
 		sb.AppendLine("\t\tprivate void InitializeComponent()");
 		sb.AppendLine("\t\t{");
-		sb.AppendLine($"\t\t\tglobal::Microsoft.Maui.Controls.Xaml.Extensions.LoadFromXaml(this, typeof({rootType}));");
+		sb.AppendLine("\t\t\t[global::System.Diagnostics.CodeAnalysis.UnconditionalSuppressMessage(\"TrimAnalysis\", \"IL2026:RequiresUnreferencedCode\",");
```

Comment:
Someone, could decorate a type with this, and I think XamlC would skip the type:

```csharp
[XamlCompilation(XamlCompilation.Skip)]
class MyPage : ContentPage
```

Should we put `[RequiresUnreferencedCode]` on this one?

https://github.com/dotnet/maui/blob/08fbc339dd492c49c01942f28412027c8a17d111/src/Controls/src/Xaml/XamlCompilationAttribute.cs#L10

Reply:
We have this issue to warn if XAML compilation is skipped: https://github.com/dotnet/maui/issues/19647. IIRC the attribute can only be applied to methods and not to enum values.